app/concerns/fetchable.rb

```diff
@@ -90,4 +91,14 @@ def downcase_scheme_and_host_and_remove_anchor_tags
   def url_extension
     Addressable::URI.parse(url).extname.downcase.from(1)
   end
+
+  def set_searchgov_domain
+    self.searchgov_domain = SearchgovDomain.find_or_create_by(domain: URI(url).host) unless url.nil?
+  end
+
+  def truncate_error_message
+    if self.last_crawl_status && self.last_crawl_status.length > 255
```

Comment:
This method can be reduced to one line. I'll give you a hint: Rail's [ActiveSupport library](http://guides.rubyonrails.org/active_support_core_extensions.html) adds some handy additional methods to core Ruby classes like `String`. Also, Ruby 2.3 introduced the '[safe navigation operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/)' to help when calling methods on a object that might be `nil`, i.e.:
```ruby
> 'foo'.reverse
"oof"
> nil.reverse
NoMethodError: undefined method `reverse' for nil:NilClass
> nil&.reverse
nil